addons/web/controllers/export.py

```diff
@@ -277,9 +277,9 @@ def _write_group_header(self, row, column, label, group, group_depth=0):
             column += 1
             aggregated_value = aggregates.get(field['name'])
             header_style = self.header_bold_style
-            if field['type'] == 'monetary':
+            if field.get('type') == 'monetary':
```

Comment:
This is probably wrong.
It is always a hazard to put get instead of [], you are just putting a "if" somewhere to avoid a crash.
Besides, fields always have a type.

The original get_field doesn't return a type for that field, I think we fall in this piece of code:
```
        if import_compat:
            if parent_field_type in ['many2one', 'many2many']:
                rec_name = Model._rec_name_fallback()
                fields = {'id': fields['id'], rec_name: fields[rec_name]}
        else:
            fields['.id'] = {**fields['id']}
```

I suppose the type is missing here and should be `#!pseudocode   ID`
@ryv-odoo @kmagusiak

Reply:
IMO, you are right @kebeclibre .
AFAIK, I used `__getitem__` to explicitly crash if the key didn't exists. In the previous version, the previous `get` actually ignored some monetary/float fields that didn't contain `type` (coming from a bug of get_fields).

Reply:
Fixed!

I am not sure what is `#!pseudocode   ID` but I applied where it was missing `type` for `{x2many_field}/id`

Reply:
The pseudo code part was just there to signal that I did not what type to place there.
The idea was to give you the hint to put the ID type `IdType` from fields.py.

Although I'm not sure how to fix.
A test would be nice too, testing the `get_fields` route

Reply:
> And fixing it in 18.0 as it only surface in this version after we decide to not use .get while accessing type. Is it okay?

Completely ok :+1: , IMO, since it was not a blocking issue and nobody complains about it.

Agree with @kebeclibre , adding a test should be great (maybe in `TestExport`).